packages/payments/src/channel-manager.ts

```diff
@@ -397,16 +397,16 @@ export class ChannelManager implements ChannelManagementAPI {
    * and then s yncs all channels for the resulting allocation
    */
   private async ensureChannelsOpen(initialMessage: Outgoing): Promise<ChannelResult[]> {
-    for (const retryTimeoutMs of [2_500, 5_000, 10_000, 20_000, 40_000]) {
-      const initialResults = await this.exchangeMessagesUntilOutboxIsEmpty(initialMessage);
+    let initialResults = await this.exchangeMessagesUntilOutboxIsEmpty(initialMessage);
 
+    for (const retryTimeoutMs of [2_500, 5_000, 10_000, 20_000, 40_000]) {
       const [running, notRunning] = _.partition(initialResults, ['status', 'running']);
-      this.logger.debug('Inserting channels into thing', {running, notRunning});
+
       if (notRunning.length === 0) return running;
 
       await delay(retryTimeoutMs);
 
-      this._syncChannels(_.map(notRunning, 'channelId'));
+      initialResults = await this._syncChannels(_.map(notRunning, 'channelId'));
```

Comment:
This overwrites the `initialResults`.

What if
1. a subset of channels become running in the first attempt
2. some other subset become stalled in the first attempt, but become running in a subsequent attempt

?

It seems as though the successful ones from (1) are now lost.

Reply:
7db8da6

Reply:
I don't know that 7db8da6 addresses this.

Say we create channels with ids `1,2,3,4,5`.

```
    let channelResults = await this.exchangeMessagesUntilOutboxIsEmpty(initialMessage);
const channelIds = _.map(channelResults, 'channelId');
```

I guess, `channelIds` should _always_ be 1,2,3,4,5, right?

Next, `channelResults` is overwritten:
```
channelResults = await this._syncChannels(
  _.difference(channelIds, _.map(running, 'channelId'))
);
```

Now, suppose `1,2,` are running. Then `_.difference(channelIds, _.map(running, 'channelId'))` would evaluate to `3,4,5` in this case.

Therefore, `channelResults` is now overwritten with _only the results_ for channels `3,4,5`.

In the next iteration of the loop, `running` would be the results for channels `3,4,5`. Therefore, `1,2` are not returned to the caller of `ensureChannelsOpen`.

Here's how I might solve this issue:
1. create a `running = new Set<ChannelResult>`.
2. After `this.exchangeMessagesUntilOutboxIsEmpty` resolves, add the running channels to `running`. Store the not running channels in `notRunning`
3. In the loop, call `_syncChannels` on the not-running channels. With the result, filter out the running channels, and add them to `running`.

But, TBH, since our plan is to "pull the guts out" here, I'm happy to intentionally leave this bug in place.

Reply:
Ah I see, improving to account for this...